### PR TITLE
Fix GetToken

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/aspnet/HttpAbstractions/issues/823

@Tratcher

### CODE REVIEW

File: src/Microsoft.AspNetCore.Authentication.Core/AuthenticationCoreServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public static IServiceCollection AddAuthenticationCore(this IServiceCollection s
                 throw new ArgumentNullException(nameof(services));
             }
 
+            services.AddOptions();
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Authentication.Core/AuthenticationCoreServiceCollectionExtensions.cs]: Hosting always adds this, no? If you're just adding it for tests then add it in the tests.

File: test/Microsoft.AspNetCore.Authentication.Core.Test/TokenExtensionTests.cs [hunk 2]
```diff
@@ -117,7 +122,68 @@ public void UpdateTokenValueReturnsFalseForUnknownToken()
             Assert.Null(props.GetTokenValue("ONE"));
             Assert.Null(props.GetTokenValue("Jigglypuff"));
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Authentication.Core.Test/TokenExtensionTests.cs]: Jigglypuff?

**Review by Tratcher — APPROVED** [review]

File: test/Microsoft.AspNetCore.Authentication.Core.Test/TokenExtensionTests.cs (hunk 2 above)
Comment (HaoK) [inline test/Microsoft.AspNetCore.Authentication.Core.Test/TokenExtensionTests.cs]: Guessing your family doen't play pokemon go :)

File: src/Microsoft.AspNetCore.Authentication.Core/AuthenticationCoreServiceCollectionExtensions.cs (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.AspNetCore.Authentication.Core/AuthenticationCoreServiceCollectionExtensions.cs]: Yup just for tests, sure I can move to there I guess

**Comment by HaoK** [comment]
3085f016e47bb3e0c90491e2c155a95b151fabda

File: test/Microsoft.AspNetCore.Authentication.Core.Test/TokenExtensionTests.cs (hunk 2 above)
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Authentication.Core.Test/TokenExtensionTests.cs]: Oh I recognized it. Testing on Pokemon is ethically discouraged, that's what Chell is for.

**Pull request closed without merging**